### ISSUE TITLE
Implement AVX-512 support

### ISSUE THREAD

**Issue by mattkretz** [VcDevel/Vc#102]
- [x] Extend build system to do AVX-512 builds
- [x] Investigate whether Travis can be instructed to run the Intel SDE for AVX-512 tests
- [ ] Investigate how large the difference to the MIC implementation would be
- [ ] depending on the above: fork/extend MIC or AVX2 to support AVX-512

**Comment by noma** [comment]
Hi, I would be highly interested in that enhancement and could offer testing. Is there a time line for this feature? 
To my experience so far, intrinsics code written for KNC SIMD instructions compiled without problems for an AVX512 (using the Intel compiler), so at least initial support should be relatively easy to accomplish.

**Comment by mattkretz** [comment]
Hi noma, thanks for your offer and info. I have not set up a development environment for KNL / Skylake-AVX512 at this point. @svalleco is working on KNL support, though. I'd be very happy for any kind of help on this front. My plate is currently full with 70% parental leave and a priority on C++ standardization...
It would be best if you and @svalleco can coordinate efforts to share the work.

**Comment by mattkretz** [comment]
AVX512 is implemented in the mkretz/datapar branch, which is to become Vc 2.0 once the P0214 proposal is merged into the next WG21 TS.
